index.d.ts

```diff
@@ -0,0 +1,52 @@
+import { AbortSignal } from "abort-controller"
+export type RandomnessBeacon = {
+    round: number
+    randomness: string
+    signature: string
+    previous_signature: string
+}
+
+export type NodeInfo = {
+    public_key: string
+    period: number
+    genesis_time: number
+    hash: string
+    groupHash: string
+    schemeID: string
+    metadata: {
+        beaconID: string
+    }
+}
+
+export type ClientOptions = {
+    chainHash: string
+    insecure?: boolean
+    disableBeaconVerification?: boolean
+    chainInfo?: string
+    cacheSize?: number
+    watcher?: Watcher
+} & WatcherOptions
+
+export type WatcherOptions = {
+    signal?: AbortSignal
+}
+
+export interface Watcher {
+    watch(options: WatcherOptions): Promise<IterableIterator<RandomnessBeacon>>
+    close(): Promise<void>
+}
+
+export class HTTP {
+    static forURLs(urls: Array<string>, chainHash: string): Promise<Array<HTTP>>
+}
+
+export class Client {
```

Comment:
This is lacking a `chains` function to get the list of chainhashes like this: http://pl-us.testnet.drand.sh/chains

Reply:
right now there isn't one in the js impl - I will raise an issue to add it though